### PR TITLE
Nexcess domains

### PR DESCRIPTION
**Pull request by jorgemasta**
I have been working on adding support o search domains using Nexcess. 

## Problem
Until now, searching for available domains within the app was impossible. Instead, the app directly redirected you to Nexcess for the whole process.

## Solution
To streamline the process, domain search and selection have been integrated into the go-live wizard. To this end, new screens have been created based on [the designs](https://www.figma.com/file/1poWoDLfFnVcFrRH3wIDyC/WME-Core?node-id=1134%3A23261).
- Find Domain. It shows a list of domains based on text input and allows you to select a domain.
- Connect with Nexcess. It shows the selected domain and a button to connect with Nexcess
- Claim Your Domain. It shows a success page

When a user clicks on "Connect Your Nexcess Account", the app makes a request to the backend that returns an URL. After receiving it, the app redirects the user to Nexcess for checkout.

After finishing the checkout process, Nexcess will redirect the user back to the app. This URL should be: `/wizard/go-live?nexcess=true&step=3&domain=jorgetri.be`. Search params are important:
- `nexcess=true` and `step=3` are used to know the state of the routes
- `domain=jorgetri.be` is used to show the user the bought domain

I have introduced some styles, but most have been reused from other existing components.

I installed [**react-query**](https://tanstack.com/query/v4/docs/adapters/react-query) to help us manage the state of the queries and mutations. It is much more powerful, but in our case, it helps us in 2 obvious cases:
- It reduces boilerplate (and thus errors) to handle the state of queries: `isLoading`, `isError`, etc.
- It cache in the client the previous queries. This gives a better UX. If, in the same session, you search for jorge.com, then change to jorge2.com and then search for jorge.com again. We will immediately show you the cached results, as 99.9% of the cases will be valid. In case of a change, they will be replaced with new data.

### Screenshots
Screenshot 2022-10-28 at 10 57 52
Screenshot 2022-10-28 at 10 58 01
Screenshot 2022-10-28 at 11 10 20

## How to test
These changes must be deployed in the WordPress plugin.

1. Open the sitebuilder
2. Open the "Go Live" wizard
3. Select "No, I need one!"
4. Search for a domain
   - the domain state can be: taken, available or selected
   - only one domain can be selected at a time
5. Select and domain and click continue
6. See a summary of the selected domain
7. Click on "Connect Your Nexcess Account"
   - It should redirect you to nexcess for checkout
   - And once the checkout is completed, be redirected to a success page in our app
8. Click Save & Exit
   - Be directed to the sitebuilder homepage

### CODE REVIEW

**Review by chumpgrub — COMMENTED** [review]
There's a lot of fantastic work here! Thank you.

I have a couple things I'd like to address before approving, however:
- This PR seems to include the solution for local development, but I've been unable to make this work and I'd like the opportunity to look at it with you and Colten.
- With the introduction of React Query to the project, I'd like an overview of the tool and how its used in this project.

File: packages/sitebuilder/sitebuilder.d.ts [hunk 1]
```diff
@@ -118,6 +118,63 @@ declare global {
...
+		metadata: DomainMetadata
+		configuration: DomainConfiguration
+		description: string
+		monthly_fee: string
+		name: string
+		term_fees: DomainTermFees
+		type: string
+	}
+
+	export interface DomainConfiguration {}
```
[the remark this answers is not in the document]
Comment (jorgemasta) [reply packages/sitebuilder/sitebuilder.d.ts:158]: Not really, removed.

**Comment by jorgemasta** [comment]
Thank you for the feedback. We can check the local environment together.

I have also updated the PR description explaining why to use react-query.

**Comment by jorgemasta** [comment]
Thank you for the feedback @chumpgrub. We can check the local environment together.

I have also updated the PR description explaining why to use react-query.

File: packages/sitebuilder/src/contexts/GoLiveProvider.tsx [hunk 2]
```diff
@@ -80,6 +81,7 @@ const GoLiveProvider = ({ children }: { children: React.ReactNode }) => {
 
 	const navigate = useNavigate();
 	const [searchParams, setSearchParams] = useSearchParams();
+	const nexcessNavigation = searchParams.get('nexcess') === 'true';
```
Comment (chumpgrub) [inline packages/sitebuilder/src/contexts/GoLiveProvider.tsx:84]: There's been a request to change the query string variable `nexcess` to `purchase`.

**Review by chumpgrub — CHANGES_REQUESTED** [review]
Small change request, please. The ream would like to change the `nexcess` query variable with `purchase`.

File: packages/sitebuilder/src/contexts/GoLiveProvider.tsx (hunk 2 above)
Comment (jorgemasta) [reply packages/sitebuilder/src/contexts/GoLiveProvider.tsx]: Done

**Review by chumpgrub — APPROVED** [review]
Thanks for this!

**Pull request merged**